### PR TITLE
Add ABOUT.md file

### PR DESCRIPTION
**Pull request by afamyial**
- Added about file for describing JavaScript.

### CODE REVIEW

**Comment by matthewmorgan** [comment]
@afamyial Looks good--  I think you should mention NodeJS.  Javascript on the server!  Maybe have a look at the ABOUT.md we hashed out for [ECMAScript](https://github.com/exercism/xecmascript/pull/115).

File: docs/ABOUT.md [hunk 1]
```diff
@@ -0,0 +1,11 @@
+# What is JavaScript?
```
Comment (kytrinyx) [inline docs/ABOUT.md]: This shows up on a page that already has a title, so it's cleaner if this is just paragraph text.

File: docs/ABOUT.md [hunk 2]
```diff
@@ -0,0 +1,11 @@
...
+---
+Javascript is a scripting language used to provide dynamic and interactive content on webpages. It does not have class-oriented inheritance, but it does have prototype-oriented inheritance. Client-side JavaScript is commonly used. JavaScript provides benefits like Less server interaction, Immediate feedback to the site visitors, Increased interactivity and Richer interfaces.
+
+# Why should one learn JavaScript?
+---
+You should learn JavaScript simply because:- 
+* Its easy to learn.
+* Its versatile.
+* Its Open Source.
+* Its universal and very much in demand.
```
Comment (kytrinyx) [inline docs/ABOUT.md]: `Its` should have an apostrophe: `It's`

File: docs/ABOUT.md [hunk 3]
```diff
@@ -0,0 +1,11 @@
+# What is JavaScript?
+---
+Javascript is a scripting language used to provide dynamic and interactive content on webpages. It does not have class-oriented inheritance, but it does have prototype-oriented inheritance. Client-side JavaScript is commonly used. JavaScript provides benefits like Less server interaction, Immediate feedback to the site visitors, Increased interactivity and Richer interfaces.
+
+# Why should one learn JavaScript?
+---
+You should learn JavaScript simply because:- 
+* Its easy to learn.
+* Its versatile.
```
Comment (kytrinyx) [inline docs/ABOUT.md]: I'm not sure what `versatile` means in this context.

File: docs/ABOUT.md [hunk 4]
```diff
@@ -0,0 +1,11 @@
+# What is JavaScript?
+---
+Javascript is a scripting language used to provide dynamic and interactive content on webpages. It does not have class-oriented inheritance, but it does have prototype-oriented inheritance. Client-side JavaScript is commonly used. JavaScript provides benefits like Less server interaction, Immediate feedback to the site visitors, Increased interactivity and Richer interfaces.
+
+# Why should one learn JavaScript?
+---
+You should learn JavaScript simply because:-
```
Comment (kytrinyx) [inline docs/ABOUT.md]: Let's remove `simply` here.

File: docs/ABOUT.md [hunk 5]
```diff
@@ -0,0 +1,11 @@
+# What is JavaScript?
+---
+Javascript is a scripting language used to provide dynamic and interactive content on webpages. It does not have class-oriented inheritance, but it does have prototype-oriented inheritance. Client-side JavaScript is commonly used. JavaScript provides benefits like Less server interaction, Immediate feedback to the site visitors, Increased interactivity and Richer interfaces.
```
Comment (kytrinyx) [inline docs/ABOUT.md]: I'm not sure what _Client-side JavaScript is commonly used._ means here.

I really like that you're talking about the benefits. How about rewording the benefits so that they're more conversational, and putting them right after the first sentence?

File: docs/ABOUT.md [hunk 6]
```diff
@@ -0,0 +1,11 @@
...
+
+# Why should one learn JavaScript?
```
Comment (kytrinyx) [inline docs/ABOUT.md]: Let's remove this header as well. The paragraph below it says it really well already.

**Comment by afamyial** [comment]
I changed the description taking care of above mentioned points. Please let me know if any more changes are required.

File: docs/ABOUT.md [hunk 7]
```diff
@@ -0,0 +1,14 @@
+Javascript is a scripting language used to provide dynamic and interactive content on webpages. It does not have class-oriented inheritance, but it does have prototype-oriented inheritance. JavaScript provides benefits like :-
```
Comment (matthewmorgan) [inline docs/ABOUT.md]: I think this first sentence should also mention the server-side uses.

File: docs/ABOUT.md [hunk 8]
```diff
@@ -0,0 +1,14 @@
+Javascript is a scripting language used to provide dynamic and interactive content on webpages. It does not have class-oriented inheritance, but it does have prototype-oriented inheritance. JavaScript provides benefits like :- 
+* Less server interaction as in validating user input before sending page to server, thus, reducing server traffic.
```
Comment (matthewmorgan) [inline docs/ABOUT.md]: This should probably read:
'JavaScript in the browser can reduce server traffic by validating user input before it is sent to the server.'

File: docs/ABOUT.md [hunk 9]
```diff
@@ -0,0 +1,14 @@
...
+* Immediate feedback to the site visitors so that they dont reload page just to find they forgot to enter something.
```
Comment (matthewmorgan) [inline docs/ABOUT.md]: I guess I don't understand this sentence.  Also, "dont" should be "don't".

File: docs/ABOUT.md [hunk 10]
```diff
@@ -0,0 +1,14 @@
...
+* Increased interactivity and Richer interfaces facilitating users to experience mouse hover changes, drag-drop components, animations etc.
```
Comment (matthewmorgan) [inline docs/ABOUT.md]: Why is "Richer" capitalized?  I would also suggest "allowing" in place of "facilitating".

File: docs/ABOUT.md [hunk 11]
```diff
@@ -0,0 +1,14 @@
...
+
+Client-Side JavaScript which enables the enhancement and manipulation of web pages and client browsers, is commonly used. The script can be referenced by or included in the HTML and code is interpreted by the browser. This takes the plain static HTML page to a level where user can interact to it.
```
Comment (matthewmorgan) [inline docs/ABOUT.md]: Maybe: "Client-side JavaScript is interpreted in the browser without requiring compilation.  This allows interactive content to be included in HTML pages."

File: docs/ABOUT.md [hunk 12]
```diff
@@ -0,0 +1,14 @@
...
+
+Server-Side JavaScript(Node.js) is another version of JavaScript that enables back-end access to databases, file systems, and servers. It is built on Google Chrome's JavaScript V8 Engine. It uses an event-driven, non-blocking I/O model that makes it lightweight and efficient. Node can be a great solution for applications requiring I/O bound operations, data streaming etc. More details can be found  [here](https://nodejs.org/en/about/)
```
Comment (matthewmorgan) [inline docs/ABOUT.md]: Server side JS isn't really "another version" of JavaScript.  It's the same language running in a different environment.  
Node is built on V8, but  JavaScript is not.  This paragraph starts out talking about Javascript but then every later reference to 'it' refers to Node.  Maybe change 'It' in the second and third sentences to 'NodeJS'.

File: docs/ABOUT.md [hunk 13]
```diff
@@ -0,0 +1,14 @@
...
+
+You should learn JavaScript because:- 
+* It's easy to learn.
+* It's versatile in the sense that it's multi-paradigm supporting including procedural, event based, object oriented and functional programming.
```
Comment (matthewmorgan) [inline docs/ABOUT.md]: The second "it's" in this sentence should be "its" -- no apostrophe.

File: docs/ABOUT.md [hunk 14]
```diff
@@ -0,0 +1,14 @@
...
+* It's Open Source.
+* It's universal and very much in demand.
```
Comment (matthewmorgan) [inline docs/ABOUT.md]: Maybe change this to "JavaScript programming skills are in high demand."

**Comment by matthewmorgan** [comment]
I appreciate the approach here-- I think you do a better job of 'selling' JavaScript and why you should use it than I did in my intro to ECMAScript!  I think the language and structure need some edits as described though.

File: docs/ABOUT.md (hunk 9 above)
Comment (afamyial) [inline docs/ABOUT.md]: I tried to mention that User can see effects of their actions without requiring to reload like seeing any error messages on form validations etc.

File: docs/ABOUT.md (hunk 12 above)
Comment (afamyial) [inline docs/ABOUT.md]: Yeah, you are right. I ll fix it.

**Comment by afamyial** [comment]
Thanks Matthew. Actually I am new to JS, so before trying it, I did a bit of research. When I visited exercism, I found that I can put that research in a concise form for JS about file. Thus, tried it. I have made the suggested changes and pushed it. Please point out the errors if you find any.

File: docs/ABOUT.md [hunk 15]
```diff
@@ -0,0 +1,14 @@
+Javascript is a scripting language used to provide dynamic and interactive content on webpages. It does not have class-oriented inheritance, but it does have prototype-oriented inheritance. JavaScript provides benefits like :- 
+* JavaScript in the browser can reduce server traffic by validating user input before it is sent to the server.
+* Immediate feedback to the site visitors so that they don't have to reload page just to get error messages on form validations.
+* Increased interactivity and richer interfaces allowing content changes on mouse hover, drag and drop gestures, and animations.
+
+Client-side JavaScript is interpreted in the browser without requiring compilation. This allows interactive content to be included in HTML pages which would be static otherwise.
```
Comment (matthewmorgan) [inline docs/ABOUT.md]: I would change this to say 'otherwise be static.'

File: docs/ABOUT.md [hunk 16]
```diff
@@ -0,0 +1,14 @@
...
+
+Server-Side JavaScript(Node.js) is JavaScript that enables back-end access to databases, file systems, and servers. NodeJS is built on Google Chrome's JavaScript V8 Engine. NodeJS uses an event-driven, non-blocking I/O model that makes it lightweight and efficient. Node can be a great solution for applications requiring I/O bound operations, data streaming etc. More details can be found  [here](https://nodejs.org/en/about/)
+
+You should learn JavaScript because:- 
+* It's easy to learn.
+* Its versatile in the sense that it's multi-paradigm supporting including procedural, event based, object oriented and functional programming.
```
Comment (matthewmorgan) [inline docs/ABOUT.md]: The first "Its" should be "It's".  The second one should be "its" rather than "it's".

File: docs/ABOUT.md [hunk 17]
```diff
@@ -0,0 +1,14 @@
+Javascript is a scripting language used to provide dynamic and interactive content on webpages. It does not have class-oriented inheritance, but it does have prototype-oriented inheritance. JavaScript provides benefits like :- 
+* JavaScript in the browser can reduce server traffic by validating user input before it is sent to the server.
```
Comment (matthewmorgan) [inline docs/ABOUT.md]: Your preceding sentence says you're going to list benefits.  I would begin each of the benefit bullet points with a verb:

1) Reducing server traffic by....
2) Providing immediate feedback...
(also "page" should probably be "pages" in the above bullet point)
3) Allowing richer user interfaces with content changes on ...

File: docs/ABOUT.md [hunk 18]
```diff
@@ -0,0 +1,14 @@
...
+* Immediate feedback to the site visitors so that they don't have to reload page just to get error messages on form validations.
+* Increased interactivity and richer interfaces allowing content changes on mouse hover, drag and drop gestures, and animations.
+
+Client-side JavaScript is interpreted in the browser without requiring compilation. This allows interactive content to be included in HTML pages which would be static otherwise.
+
+Server-Side JavaScript(Node.js) is JavaScript that enables back-end access to databases, file systems, and servers. NodeJS is built on Google Chrome's JavaScript V8 Engine. NodeJS uses an event-driven, non-blocking I/O model that makes it lightweight and efficient. Node can be a great solution for applications requiring I/O bound operations, data streaming etc. More details can be found  [here](https://nodejs.org/en/about/)
```
Comment (matthewmorgan) [inline docs/ABOUT.md]: Again, Node is not JavaScript.  This should probably be reworded:

> Server-side JavaScript as run in NodeJS enables back-end access....

**Comment by matthewmorgan** [comment]
@afamyial I'm suggesting a few more stylistic and grammatical changes.  

Also, if you can squash your commits into one commit, that's the usual way for PRs here.  Please let me know if you need help doing that.  I would be happy to assist.

**Comment by afamyial** [comment]
@matthewmorgan I squashed the commits and pushed. In case any other change required, please let me know.

File: docs/ABOUT.md [hunk 19]
```diff
@@ -0,0 +1,14 @@
+Javascript is a scripting language used to provide dynamic and interactive content on webpages. It does not have class-oriented inheritance, but it does have prototype-oriented inheritance. Also, server side JS allows to use same language on server and client besides being fast. JavaScript provides benefits like :-
```
Comment (matthewmorgan) [inline docs/ABOUT.md]: The third sentence seems to have a few issues.  Should read perhaps "allows one to use the same language on the server and client.  Besides being fast, JavaScript provides..."

File: docs/ABOUT.md [hunk 20]
```diff
@@ -0,0 +1,14 @@
...
+* Providing immediate feedback to the site visitors so that they don't have to reload pages just to get error messages on form validations.
+* Allowing richer user interfaces with content changes on mouse hover, drag and drop gestures, and animations.
+
+Client-side JavaScript is interpreted in the browser without requiring compilation. This allows interactive content to be included in HTML pages which would otherwise be static.
+
+Server-Side JavaScript as run in NodeJS enables back-end access to databases, file systems, and servers. NodeJS is built on Google Chrome's JavaScript V8 Engine. NodeJS uses an event-driven, non-blocking I/O model that makes it lightweight and efficient. Node can be a great solution for applications requiring I/O bound operations, data streaming etc. More details can be found  [here](https://nodejs.org/en/about/)
+
+You should learn JavaScript because:- 
+* It's easy to learn.
+* It's versatile in the sense that its multi-paradigm supporting including procedural, event based, object oriented and functional programming.
```
Comment (matthewmorgan) [inline docs/ABOUT.md]: I may have made a mistake in my feedback here.  Since the diff is gone I can't tell, but this should probably read:

"sense that it's multi-paradigm..."  You want the apostrophe in that "it's".

**Comment by matthewmorgan** [comment]
@afamyial a couple of minor additional changes.  

@kytrinyx would you care to comment?

**Comment by kytrinyx** [comment]
This looks really good.

Let's delete the second sentence: _It does not have class-oriented inheritance, but it does have prototype-oriented inheritance._ (this is covered in the "versatile" bullet point further down).

Other than this, just that one typo (its vs it's).

**Comment by afamyial** [comment]
Fixed everything. :)

File: docs/ABOUT.md [hunk 21]
```diff
@@ -0,0 +1,14 @@
...
+* Providing immediate feedback to the site visitors so that they don't have to reload pages just to get error messages on form validations.
+* Allowing richer user interfaces with content changes on mouse hover, drag and drop gestures, and animations.
+
+Client-side JavaScript is interpreted in the browser without requiring compilation. This allows interactive content to be included in HTML pages which would otherwise be static.
+
+Server-Side JavaScript as run in NodeJS enables back-end access to databases, file systems, and servers. NodeJS is built on Google Chrome's JavaScript V8 Engine. NodeJS uses an event-driven, non-blocking I/O model that makes it lightweight and efficient. Node can be a great solution for applications requiring I/O bound operations, data streaming etc. More details can be found  [here](https://nodejs.org/en/about/)
+
+You should learn JavaScript because:- 
+* It's easy to learn.
+* It's versatile in the sense that it's multi-paradigm supporting including procedural, event based, object oriented and functional programming.
```
Comment (matthewmorgan) [inline docs/ABOUT.md]: I would omit 'including' from this line.  It doesn't make sense here.

File: docs/ABOUT.md [hunk 22]
```diff
@@ -0,0 +1,14 @@
+Javascript is a scripting language used to provide dynamic and interactive content on webpages. Also, server side JS allows to use same language on server and client. Besides being fast, JavaScript provides benefits like :-
```
Comment (matthewmorgan) [inline docs/ABOUT.md]: change to 'allows the use of the same language on the server'.

**Comment by matthewmorgan** [comment]
@afamyial if you don't mind making the last couple of edits I mentioned I think this is ready to merge.

Thanks!

**Comment by afamyial** [comment]
@matthewmorgan I have made the changes and pushed as one commit.

**Comment by matthewmorgan** [comment]
Awesome!  Thanks.

**Comment by afamyial** [comment]
@matthewmorgan It wasn't a problem at all. Infact, I felt nice and learnt a lot. Thanks for the feedbacks you and Katrina provided. Really helped a lot.

**Comment by rchavarria** [comment]
Thanks @afamyial. Wow, it hasn't been easy. Natural language is not as precise as a programming language and it's difficult to find the right words.

Thanks for this great contribution. I'd also like to invite you to take a look at the [ECMAScript track](http://exercism.io/languages/ecmascript). It's also for programming in JavaScript, but you can use the latest functionalities brought into JavaScript.

**Comment by afamyial** [comment]
@rchavarria Sure, I'll take a look into it.
